features/conversation/conversation-message/conversation-message-content-types/conversation-message-reply.tsx

```diff
@@ -109,144 +118,178 @@ const MessageReplyReference = memo(function MessageReplyReference(props: {
+
+  const topic = useCurrentConversationTopic();
+
+  const { message: replyMessageReferenceMessage } = useConversationMessageById({
+    messageId: referenceMessageId,
+    topic,
+  });
 
   const inboxName = usePreferredInboxName(
-    replyMessageReference?.senderAddress as InboxId
+    replyMessageReferenceMessage?.senderAddress as InboxId
   );
 
+  const tapGesture = Gesture.Tap()
+    .onBegin(() => {
+      conversationStore.setState({
+        highlightedMessageId: referenceMessageId,
+        scrollToMessageId: referenceMessageId,
+      });
+    })
+    .runOnJS(true);
+
   return (
-    <VStack
-      style={{
-        rowGap: theme.spacing.xxxs,
-        flex: 1,
-        backgroundColor: fromMe
-          ? theme.colors.bubbles.nestedReplyFromMe
-          : theme.colors.bubbles.nestedReply,
-        borderRadius:
-          theme.borderRadius.message.bubble -
-          theme.spacing.message.replyMessage.horizontalPadding / 2, // / 2 so the border fits the border radius of BubbleContentContainer
-        paddingHorizontal: theme.spacing.xs,
-        paddingVertical: theme.spacing.xxs,
-      }}
-    >
-      <HStack
+    <GestureDetector gesture={tapGesture}>
+      <VStack
         style={{
-          alignItems: "center",
-          columnGap: theme.spacing.xxxs,
+          rowGap: theme.spacing.xxxs,
+          flex: 1,
+          backgroundColor: fromMe
+            ? theme.colors.bubbles.nestedReplyFromMe
+            : theme.colors.bubbles.nestedReply,
+          borderRadius:
+            theme.borderRadius.message.bubble -
+            theme.spacing.message.replyMessage.horizontalPadding / 2, // / 2 so the border fits the border radius of BubbleContentContainer
+          paddingHorizontal: theme.spacing.xs,
+          paddingVertical: theme.spacing.xxs,
         }}
       >
-        <Icon
-          size={theme.iconSize.xs}
-          icon="arrowshape.turn.up.left.fill"
-          color={
-            fromMe
-              ? theme.colors.text.inverted.secondary
-              : theme.colors.text.secondary
-          }
-        />
-        <Text preset="smaller" color="secondary" inverted={fromMe}>
-          {inboxName}
-        </Text>
-      </HStack>
-      {!!replyMessageReference && (
-        <MessageReplyReferenceContent replyMessage={replyMessageReference} />
-      )}
-    </VStack>
+        <HStack
+          style={{
+            alignItems: "center",
+            columnGap: theme.spacing.xxxs,
+          }}
+        >
+          <Icon
+            size={theme.iconSize.xs}
+            icon="arrowshape.turn.up.left.fill"
+            color={
+              fromMe
+                ? theme.colors.text.inverted.secondary
+                : theme.colors.text.secondary
+            }
+          />
+          <Text preset="smaller" color="secondary" inverted={fromMe}>
+            {inboxName}
+          </Text>
+        </HStack>
+        {!!replyMessageReferenceMessage && (
```

Comment:
this replyMessageReferenceMessage is a little confusingly named - not sure what it is exactly or why not just replyMessage?

Reply:
renamed to `referencedMessage` since it's the message associated with the reply

Reply:
It's the reference message that the actual message replied to